nextcord/application_command.py

```diff
@@ -1265,77 +1305,176 @@ def __init__(
+                    parameter.annotation, list(self.option_types.keys())
+                )
+            # Make sure that all literals are only OptionConverters and nothing else.
+            for lit in literals:
+                if not isinstance(lit, OptionConverter) or lit is not None:
+                    raise ValueError(
+                        f"{self.error_name} You cannot use non-OptionConverter literals when the base annotation is "
+                        f"not Literal."
+                    )
+
+            # Pyright gets upset at appending these lists together, but not upset when .extend is used?
+            # grouped_annotations: List[Union[type, Annotated[object, OptionConverter]]] = unpacked_annotations + \
+            #                                                                              literals
+            grouped_annotations: List[
+                Union[type, Annotated[Optional[OptionConverter], object], Type[None]]
+            ] = []
+            grouped_annotations.extend(unpacked_annotations)
+            grouped_annotations.extend(literals)
+            # The only literals in this should be OptionConverters. Anything else should have triggered the ValueError.
+            for anno in grouped_annotations:
+                if isinstance(anno, object) and isinstance(anno, OptionConverter):
+                    # If the annotation is instantiated, add it to the converters and set the anno to the type it has.
+                    annotation_converters.append(anno)
+                    anno = anno.type
+                elif isinstance(anno, type) and issubclass(anno, OptionConverter):
+                    # If the annotation is NOT instantiated, instantiate it and do the above.
+                    made_converter = anno()
+                    annotation_converters.append(made_converter)
+                    anno = made_converter.type
+
+                if anno is None or anno is type(None):
+                    # If None is included, they want it to be optional. But we don't want None processed fully as anno.
+                    annotation_required = False
+                else:
+                    if found_type is MISSING:
+                        # If we haven't set the type of the annotation, set it.
+                        found_type = self.get_type(anno)
+                    elif self.get_type(anno) is not found_type:
+                        raise ValueError(
+                            f"{self.error_name} | Annotation {anno} is incompatible with {found_type} \n| {typehint_origin}\n| {parameter.annotation}\n| {grouped_annotations}"
+                        )
+
+                    # Theoretically, it COULD be an OptionConverter at this point if it somehow skipped passed the upper
+                    #  two isinstance if statements AND didn't error on self.get_type.
+                    if isinstance(anno, ApplicationCommandOptionType) or isinstance(
```

Comment:
This should just be combined with the below `elif` by negating it
```py
if foo:
    pass
elif bar:
    code()
```
vvv
```py
if not foo and bar:
    code()
```

Reply:
Fiixed!